### PR TITLE
Expose Reaper Metrics through a ServiceMonitor (K8SSAND-1587)

### LINKED ISSUE #578: K8SSAND-1587 ⁃ Expose Reaper Metrics through a ServiceMonitor
A [recent contribution to Reaper](https://github.com/thelastpickle/cassandra-reaper/commit/44574c2bdcac7b8742098e7617a31072bbdda0f2) added a Prometheus Metrics endpoint which we can leverage through a ServiceMonitor in K8ssandra-operator.

This would require the addition of a new `Telemetry` setting in the K8ssandraCluster and Reaper CRDs to enable metrics scraping, after an upgrade to the latest Reaper release was made.

This would also requiring exposing the port of the Prometheus endpoint in the Reaper pod and service.

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1587) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1587
┆priority: Medium

### PR DESCRIPTION
**Pull request by emerkle826**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #578 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by adejanovski — CHANGES_REQUESTED** [review]
Looks good so far. There's one copy paste issue and I need to run some deployment tests before I can approve it.
Seems like we have no e2e tests for telemetry btw 😕 Let's keep it out of the scope of this PR though.

**Comment by adejanovski** [comment]
Could you upgrade Reaper to 3.2.0 in this PR @emerkle826 and add it to the changelog?
Thanks!

File: controllers/reaper/reaper_telemetry_reconciler.go
Comment (emerkle826) [reply controllers/reaper/reaper_telemetry_reconciler.go:76]: good catch. I figured I was going to have a copy-paste issue.

**Comment by emerkle826** [comment]
> Could you upgrade Reaper to 3.2.0 in this PR @emerkle826 and add it to the changelog? Thanks!

will do

**Comment by emerkle826** [comment]
Got a failing test locally again. Hold off on merge.

File: controllers/reaper/reaper_telemetry_reconciler.go
Comment (adejanovski) [inline controllers/reaper/reaper_telemetry_reconciler.go:32]: This function is never invoked, which results in no service monitor being created.
You need to invoke it in the Reaper controller, like it's done for Stargate in the Stargate controller.
It does seem like we have a code coverage issue as well since this wasn't picked up by the tests.

**Review by adejanovski — CHANGES_REQUESTED** [review]
I'm not sure how easy it is, but we'd need an e2e test to perform basic checks on the presence of ServiceMonitor objects if they are requested.
It will require installing the prometheus operator CRDs prior to the K8ssandraCluster object creation.
@Miles-Garnsey, wdyt?

File: controllers/reaper/reaper_telemetry_reconciler.go
Comment (emerkle826) [reply controllers/reaper/reaper_telemetry_reconciler.go:32]: You're right, I totally missed that. I'll get it fixed and I'm working on an e2e test to verify the service monitor bits.

Comment (jsanda) [reply controllers/reaper/reaper_telemetry_reconciler.go:32]: Are we planning to install Prometheus for this e2e test? If not, there isn't any benefit in doing an e2e test vs an envtest integration test. Checking the presence of the ServiceMonitor is precisely the type of thing we want to do in envtests.

Comment (Miles-Garnsey) [reply controllers/reaper/reaper_telemetry_reconciler.go:32]: Would you like to maybe just hook into my existing tests over [here](https://github.com/k8ssandra/k8ssandra-operator/tree/main/test/kuttl/test-servicemonitors)? That'll get you everything you need with ~20 lines of additional config.

Comment (adejanovski) [reply controllers/reaper/reaper_telemetry_reconciler.go:32]: Hooking into existing tests seems like the best thing to do 👍 

@emerkle826 @Miles-Garnsey, my understanding of these tests is that [this kustomization](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/kuttl/test-servicemonitors/config/single-node-k8ssandra/kustomization.yaml) needs to be modified to add reaper with the telemetry block, and then update [this file](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/kuttl/test-servicemonitors/03-assert.yaml) to add the assertion for the new service monitor, right?

Comment (Miles-Garnsey) [reply controllers/reaper/reaper_telemetry_reconciler.go:32]: Yes, absolutely correct.

**Review by adejanovski — APPROVED** [review]
Looks good now @emerkle826 ✅ 
Thanks for adding the tests!
:shipit:

**Pull request merged**
